src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -450,6 +450,16 @@ public void filterSensitiveInformation() {
         super.filterSensitiveInformation();
     }
 
+    /**
+     * Check if manual results are allowed for the exercise
+     * @return true if manual results are allowed, false otherwise
+     */
+    public boolean areManualResultsAllowed() {
```

Comment:
Just for documentation: What happens if you postpone the due date? As I was testing your changes I postponed the due date and afterwards, I was able to assess submissions although the due date did not pass yet and students are still able to submit new solutions. We should either disable changing the due date or disable assessing if the due date changes. I think this should be handled in a separate PR.
Also from the students' side 
![image](https://user-images.githubusercontent.com/50635005/70153210-de6c9500-16ae-11ea-9fe4-9c2da644d42d.png)
the due date is messed up as they can submit new results even though the due date has passed. But as I said this might be out of this PRs scope